twilly_cli/src/sync/maps.rs

```diff
@@ -80,6 +89,187 @@ pub async fn choose_map_action(twilio: &Client, sync_service: &SyncService) {
                     println!("{:#?}", selected_sync_map);
                     println!();
                 }
+                Action::Rename => {
+                    let get_name_prompt = Text::new(
+                        "What would you like to rename this map to? Must be supported characters '^[a-zA-Z0-9-_]+$'"
+                    );
+                    let get_name_result = prompt_user(get_name_prompt);
+
+                    if let None = get_name_result {
+                        break;
+                    }
+
+                    let allowed_chars = Regex::new(r"^[a-zA-Z0-9-_]+$").unwrap();
```

Comment:
This validation logic could be moved inside a custom `Text` validator. See the [documentation here](https://github.com/mikaelmello/inquire?tab=readme-ov-file#validation) & an [example from this repo](https://github.com/TristanBlackwell/twilly/blob/2d24dd746771b5cb563d23b65ca4e5213572acd4/twilly_cli/src/account.rs#L35C25-L40C27).

This will display the error immediately (in red) and gives the user opportunity to keep trying until they cancel the action.

Reply:
Resolved 👍